homeassistant/components/dremel_3d_printer/sensor.py

```diff
@@ -0,0 +1,263 @@
+"""Support for monitoring Dremel 3D Printer sensors."""
+from __future__ import annotations
+
+from collections.abc import Callable
+from dataclasses import dataclass
+
+from dremel3dpy import Dremel3DPrinter
+
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+    SensorStateClass,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    PERCENTAGE,
+    EntityCategory,
+    UnitOfInformation,
+    UnitOfTemperature,
+    UnitOfTime,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.typing import StateType
+
+from .const import ATTR_EXTRUDER, ATTR_PLATFORM, DOMAIN
+from .entity import Dremel3DPrinterEntity
+
+
+@dataclass
+class Dremel3DPrinterSensorEntityMixin:
+    """Mixin for Dremel 3D Printer sensor."""
+
+    value_fn: Callable[[Dremel3DPrinter, str], StateType]
+
+
+@dataclass
+class Dremel3DPrinterSensorEntityDescription(
+    SensorEntityDescription, Dremel3DPrinterSensorEntityMixin
+):
+    """Describes a Dremel 3D Printer sensor."""
+
+
+SENSOR_TYPES: tuple[Dremel3DPrinterSensorEntityDescription, ...] = (
+    Dremel3DPrinterSensorEntityDescription(
+        key="job_phase",
+        name="Job phase",
+        icon="mdi:printer-3d",
+        value_fn=lambda api, _: api.get_printing_status(),
+    ),
+    Dremel3DPrinterSensorEntityDescription(
+        key="remaining_time",
+        name="Remaining time",
+        native_unit_of_measurement=UnitOfTime.SECONDS,
+        device_class=SensorDeviceClass.DURATION,
+        value_fn=lambda api, key: api.get_job_status()[key],
```

Comment:
Use the `ignore_variance` helper to convert this to a time stamp. See:

https://github.com/home-assistant/core/blob/01c82131c4754097f415fe154ac0c1844837239d/homeassistant/components/prusalink/sensor.py#L145-L146

Reply:
The only thing I notice is this displays time in the past. Can we return `None` when the `datetime` is in the past?

Reply:
Still working on the last bit with the wrapper. I don't wrap everyday. 😄